mitiq/shadows/classical_postprocessing.py

```diff
@@ -323,39 +323,36 @@ def expectation_estimation_shadow(
     for qubit, pauli in obs.items():
         target_obs.append(str(pauli))
         target_locs.append(int(qubit))
-    # if target_locs is [2,3,4], then target_support is "001110"
-    target_support = create_string(num_qubits, target_locs)
 
     # classical values stored in classical computer
-    b_lists_shadow = np.array(measurement_outcomes[0])
-    u_lists_shadow = np.array([list(u) for u in measurement_outcomes[1]])
-    # number of measurements in each split
-    n_total_measurements_shadow = len(b_lists_shadow)
+    b_lists_shadow = np.array([list(u) for u in measurement_outcomes[0]])[
+        :, target_locs
+    ]
+    u_lists_shadow = np.array([list(u) for u in measurement_outcomes[1]])[
+        :, target_locs
+    ]
+
     means = []
+
     # loop over the splits of the shadow:
-    group_idxes = np.array_split(
-        np.arange(n_total_measurements_shadow), k_shadows
-    )
+    group_idxes = np.array_split(np.arange(len(b_lists_shadow)), k_shadows)
+
     # loop over the splits of the shadow:
     for idxes in group_idxes:
-        b_lists_shadow_k = b_lists_shadow[idxes]
-        u_lists_shadow_k = u_lists_shadow[idxes]
-        # number of measurements/shadows in each split
-        n_group_measurements = len(b_lists_shadow_k)
-
-        indices = np.all(
-            u_lists_shadow_k[:, target_locs] == target_obs, axis=1
-        )
-        if sum(indices) == 0:
-            means.append(0.0)
-        else:
-            eigenvalues = np.array(
-                [
-                    bitstring_to_eigenvalues(b)
-                    for b in b_lists_shadow_k[indices]
-                ]
+        if len(
+            np.nonzero(np.all(u_lists_shadow[idxes] == target_obs, axis=1))[0]
+        ):
+            product = (-1) ** np.mod(
```

Comment:
Maybe the `np.mod()` function here is redundant ?  If I am wrong ignore this comment.